--- a/target/linux/ipq40xx/image/p2w_r619ac-128m.dts
+++ b/target/linux/ipq40xx/image/p2w_r619ac-128m.dts
@@ -12,8 +12,6 @@
 		compatible = "p2w,r619ac-128m", "qcom,ipq4019";
 		model = "P&W R619AC (128M)";
 
-		qcom,ath10k-calibration-virtual = [00000000 ... ];
-		local-mac-address = [00 00 00 00 00 00];
 
 		aliases {
 			serial0 = &blsp1_uart1;
@@ -23,6 +21,24 @@
 		};
 	};
 
+	art: art@100000 {
+		compatible = "qcom,msm-part";
+		reg = <0x100000 0x10000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		status = "okay";
+
+		caldata_wifi0: caldata@0 {
+			reg = <0x0 0x1000>;
+		};
+		caldata_wifi1: caldata@1000 {
+			reg = <0x1000 0x1000>;
+		};
+		macaddr_wifi0: mac@6 {
+			reg = <0x6 0x6>;
+		};
+		macaddr_wifi1: mac@c {
+			reg = <0xc 0x6>;
+		};
+	};
+
 	soc {
 		rng@22000 {
 			status = "okay";
@@ -77,6 +93,10 @@
 	};
 
 	wifi0 {
+		nvmem-cells = <&caldata_wifi0>, <&macaddr_wifi0>;
+		nvmem-cell-names = "calibration", "mac-address";
+	};
+	wifi1 {
+		nvmem-cells = <&caldata_wifi1>, <&macaddr_wifi1>;
 		status = "okay";
 	};
 };
